crates/polars-arrow/src/array/binview/mod.rs

```diff
@@ -159,6 +159,34 @@ impl<T: ViewType + ?Sized> BinaryViewArrayGeneric<T> {
         total_bytes_len: usize,
         total_buffer_len: usize,
     ) -> Self {
+        // Verify the invariants
+        #[cfg(debug_assertions)]
+        {
+            // @TODO: Enable this. This is currently bugged with concatenate.
```

Comment:
Is our information incorrect? Where does this happen? Should I fix this is in a seperate PR?

Reply:
This is happening in the `legacy::concatenate` kernel.